### PR TITLE
Add year functionality.

### PR DESCRIPTION
**Pull request by kayernyc**
Added basic year functionality to date picker.

### CODE REVIEW

File: src/views/YearsViews.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  constructor(model: DatePickerFactory, viewConfiguration: ViewConfiguration, grid: boolean) {
+    const continuousScroll = true;
+
+    // Until date ranges are implemented, looping is not possible with years.
+    // if (viewConfiguration.looping === true || viewConfiguration.continuousScroll === undefined) {
+    //   continuousScroll = false
+    // }
+    super(model, continuousScroll);
+
+    grid ? this.frameElementClassName = 'date-picker-grid' : this.frameElementClassName = 'date-picker-list';
```
Comment (deadnancy) [inline src/views/YearsViews.ts:22]: Less redundant: `this.frameElementClassName = grid ? 'date-picker-grid' : 'date-picker-list';`.

File: src/DatePickerControl.ts [hunk 2]
```diff
@@ -101,6 +104,12 @@ export default class DatePickerControl {
           view = new WeekView(viewModel, viewConfiguration, (viewType === ViewType.GRID ? true : false));
           break;
 
+        case dateType === DateType.YEAR && viewType === ViewType.LIST:
+        case dateType === DateType.YEAR && viewType === ViewType.GRID:
+          // Looping is not possible with years until date ranges are implemented.
+          view = new YearsView(viewModel, viewConfiguration, (viewType === ViewType.GRID ? true : false));
```
Comment (deadnancy) [inline src/DatePickerControl.ts:110]: The ternary isn't really necessary. `(viewType === ViewType.GRID)` will work.

File: src/views/YearsViews.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    if (this.virtualDom) {
+      // DatePickerBaseView has determined that a virtualDom is needed
+      this.virtualDom.buildView(array, frameElement);
+      return
+    }
+
+    this.buildDateView(array);
+  }
+}
```
Comment (deadnancy) [inline src/views/YearsViews.ts:49]: image

**Review by deadnancy — APPROVED** [review]
Tiny issues but no blockers.

File: src/views/YearsViews.ts (hunk 1 above)
Comment (kayernyc) [reply src/views/YearsViews.ts:22]: excellent call!

**Pull request merged**